### ISSUE TITLE
GraphQL client not taking available credentials

### ISSUE THREAD

**Issue by Seb-Solon** [Shopify/shopify_python_api#372]
Hi there,

I was playing with the brand new GraphQL client embedded in the lib but I noticed that it was not taking the credentials. 

Even the Readme suggest that to use the GraphQL step 1-5 are required (getting a token with oath) to work, the doc says otherwise : https://shopify.dev/docs/admin-api/graphql/getting-started 
> Include the access token as a X-Shopify-Access-Token header in all GraphQL queries and mutations

Non working example: 

```python3

import shopify

API_KEY = "mykey"
PASSWORD = "mypassword"
website = "mywebsite"

shopify.ShopifyResource.set_site("https://%s:%s@%s.myshopify.com/admin/api/2020-01" % ( API_KEY, PASSWORD, website))

# returns HTTPError: HTTP Error 401: Unauthorized
shopify.GraphQL().execute("{ shop { name id } }")
```
A simple fix outside of the lib is the following : 
```python3
# manual fix . shopify.ShopifyResource.password is set by the set_url previous call
shopify.ShopifyResource.headers.update({'X-Shopify-Access-Token': shopify.ShopifyResource.password })

# returns expected data
shopify.GraphQL().execute("{ shop { name id } }")
```

This does not happen if we create a shopify session and use it :

```python3
session = shopify.Session("%s.myshopify.com" % website, "2020-01", PASSWORD)
shopify.ShopifyResource.activate_session(session)
```

I believe this should be patched in the lib itself at  is an easy one.
Adding the line : 
```python3
self.headers.update({'X-Shopify-Access-Token': shopify.ShopifyResource.password })
```
after https://github.com/Shopify/shopify_python_api/blob/master/shopify/resources/graphql.py#L10 seems to be enough. 

I can do the PR if you want even if I believe it might take longer to wait for my PR than committing the fix :wink: 

Regards

**Comment by ThomasDeBonnet** [comment]
Thanks man, made my day!

**Comment by racket99** [comment]
Thank you. This has been plaguing me for weeks!

**Comment by tanema** [comment]
Also this can be done with smaller code with a temp session
```python
with shopify.Session.temp('can-i-buy-a-feeling.myshopify.com', '2020-01', 'access_token'):
    shopify.GraphQL().execute("{ shop { name id } }")
```

**Comment by Seb-Solon** [comment]
@tanema I think you misunderstood the point of my ticket here : reporting a unexpected behavior of the API regarding the provided documentations (the one in the README, python specific, and the generic one), not looking for a way to make it work. 

I mentioned in the initial message how I made it work in various ways and `activate_session` was part of them. I suggested an fix in the code rather than the README but it's up to you. 

The step 2 of getting started mentions `For a private App you just need to set the base site url` and `That's it you're done, skip to step 6 and start using the API` . This is partially true here as it's one (basic auth) of the 2 possible ways (basic auth or acces_token) for a private app : https://shopify.dev/tutorials/authenticate-a-private-app-with-shopify-admin

Then, for the graphQL the README mentions `The authentication process (steps 1-5 under Getting Started) is identical` which is wrong, in the case of Private app, because it only support the acces_token authentication. In such case, api_token and password are the same value so taking the token value from the password field without any line of code make sense.

So we have 2 options here : 
 * The README is fixed by removing the set_site call completely (and the function itself?) and aligning the `session_activate` for private, custom and public apps. Dev would only create session and activate them, which work for Admin API and GraphQL API.

 * The set_site (basic auth) function is fixed to update the headers with the access_token. This function is only used for private app from what I understand. It is better than what I suggested earlier in the GraphQL class, as we don't know if we use a private app in this context.
 
Again, up to you for the fix implementation, but if we leave it as is people might get confused (see previous comments on that issue)
